### ISSUE TITLE
As a user, I want to see a large version of a single card image with associated events so that I can see the events in the context of the archival source.

### ISSUE THREAD

**Issue by rlskoeser** [Princeton-CDH/mep-django#248]
# testing notes
## design links
[desktop](https://zpl.io/VqkLYgN) / [tablet](https://zpl.io/an7LkRJ) / [mobile](https://zpl.io/29zjwvW)

you can get here by going to a member's "cards" section (described on #247) and clicking any of the cards in the gallery. @rlskoeser and I have been testing with [this stein card from 1923-24](https://test-shakespeareandco.cdh.princeton.edu/members/stein-gertrude/cards/50c6c543-8f19-47db-af35-f43f309fa0ec/), but you can try others too.

## checklist
- should look like the designs, on all screen types (desktop/tablet/mobile)
- card picture should be the right size and placement
- arrows on either side of the card should be clickable if there is a next/previous card to view
- if there's no next/previous card, that arrow should be greyed out and unclickable
- events in the table should match the events on the card
- table styles should change to the list layout on mobile
- on desktop, the table headers should stick to the top of the page when you scroll
- the breadcrumbs should show the dates of the card you're viewing

note there are a few discrepancies from the design; on mobile item titles will be italicized and the type of the event (e.g. Borrow) will be lowercase to create more contrast.

## dev notes
Accessible from member detail page or card image search/browse. Card Image Viewer in Zeplin.

- no deep zoom
- pagination is tracked on #249 (including thumbnail display)
- table is not sortable
- table converts to tiles on mobile (similar to activity pages)
- events need to be found based on footnotes associated with the canvas, and should be displayed in start date order
- breadcrumb label is year or date range
- image alt text: summarize date range of events and refer to table

**Comment by jkotin** [comment]
This looks great! I have a number of comments—some may be more relevant than others.

1. For Stein (following the link provided in the design links), there are still Leo Stein cards present with Gertrudes. @clmahoney is this only a test site issue?

2. In the table, I don't think the titles should be centred. Could they be left justified like the other columns?

3. In the table, would it be possible to include the publication date with the book title? The titles won't ever be exact transcriptions of the titles on the cards anyway. (They will be the corrected titles from the database.) It would be nice to have the dates too.

4. The card image is good. Just to ask: would it be possible to make them life-size on laptop and desktop? 12 cm x 17 cm?

5. Should we discuss what we want to do with the blank card? Are we sure that the we are including every blank verso? Should we include a note explaining why there are blank cards? Should we delete the blank cards? @clmahoney could we discuss these questions at an upcoming meeting?

6. I like how the card images save when right-click, save-as. Do we want to change the default file name?

But the basic functionality is perfect. @clmahoney would you test (including on mobile and tablet) and report back if you think the issue can be closed?

**Comment by jkotin** [comment]
Looking back over the exact issue to test, I only have one relevant comment: could the images be bigger?  If not, what we have works.

**Comment by clmahoney** [comment]
@jkotin No, unfortunately the Figgy items I requested be fixed by the Figgy team have not been fixed yet. I will email Kim tomorrow to ask her the status on this part of the project.

**Comment by clmahoney** [comment]
Tablet: tested on Antoine-May  
- YES, when I click on an individual card in the card gallery, the design looks like the Zeplin design
- YES, the card picture looks like it's the right size and is in the right place
- YEs, I can click the arrows on either side of the card to take me through the many cards Antoine-May has. When I get to the end of the cards, the arrow is greyed out and not clickable. 
- YES, the events in the table match the events on the card. If it's a borrow, it says borrow. If the book was crossed out, there is an "-" in the type field. 
- YES, the breadcrumbs change to show the dates of the card I'm viewing.

**Comment by clmahoney** [comment]
Mobile: tested on Sylvia Beach
- YES, for the most part this design matches the Mobile design on Zeplin. The only thing I can see that is different is that the type of event (borrow, in Beach's case on s1) is not capitalized in the text box, and it should be. See screenshot: 
![Image from iOS (5)](https://user-images.githubusercontent.com/16673934/69918167-a100d100-143c-11ea-96aa-e1046c5de060.png)

- YES, the card picture looks like it's the right size and is in the right place
- YES, I can click on the arrows on either side of the card and it takes me to the next card. When I reach the end of the cards, the arrow is greyed out and unclickable. 
- YES, the events in the text table match the events on the card 
- YES, the breadcrumbs change to match the dates of the card I've clicked on. So for Beach, s1 is 1937-40, which the breadcrumb matches, and then if I go to s2 which is blank, the breadcrumb says "unknown" (which is correct)

**Comment by rlskoeser** [comment]
@gissoo please respond to Josh's comments about left justifying titles, adding publication dates, and making the images bigger.

@jkotin responses to other comments:
4. My concerned about making the images bigger is that even at the current size, on some screens it may not be obvious that there is content below the image. Making them larger will make this worse.
5. We could programmatically exclude cards with no footnotes - this will skip over blank cards, but also possibly skip over cards that are not blank but just don't have any documented events in the system. Is that acceptable? Also, what's the priority of skipping unknown cards? Normally I would say that's a new feature request, but if you think it's important to have before we make the members live we could fold it into this story.
6. The images are being served out by the library, I don't think we have any control over the default filename when you download.

@clmahoney thanks for catching that discrepancy in the design. I think we made the borrow lower-case here to match how we were handling it on the book activity list (which I think we did to help distinguish the titles more). @gissoo we should keep these consistent, right?

**Comment by gissoo** [comment]
@jkotin @rlskoeser 
1. Left justifying titles: I agree with this, they would look more readable than being centered here, even though the logic will be inconsistent with the activities table, it will be more readable. 
2. Adding publication dates: I agree with adding publication dates to the right of the titles column on desktop and tablet, and on mobile to be on the second line from the top below the title written as "published 1921", for instance, so that it would read like a sentence. 
3. Making images bigger: I would highly recommend we do _**not**_ do this. Making them larger will make it harder to view the page and the information below, it brings usability down as they are readable at the current size. 
4. @clmahoney The "borrow" capitalization: The term, "borrow" needs to be in lowercase on mobile, it is now matching the current design on Zeplin, it is designed this way to match how we are handling the book activity, for the goal of easier readability – to be read like a sentence. Thank you for catching the inconsistency! Fixed on Zeplin now.

**Comment by gissoo** [comment]
Edit: @clmahoney @rlskoeser @jkotin 
4. Capitalizing "borrow": Since this terms has changed from "borrowed" to "borrow", I would suggest capitalizing it everywhere now, it anyway doesn't read like a sentence anymore since it's not a verb anymore.

**Comment by rlskoeser** [comment]
@gissoo we could pretty easily switch it to borrowed and purchased (which I do think would look and read better).  There was a question about labeling generic events, but it sounds like that's something to be handled later. Did we ever resolve the question of what to label that column? The label and the form of the word seem related.

**Comment by gissoo** [comment]
@rlskoeser Let's discuss this at our meeting, this is something that @jkotin also wanted to discuss at the meeting rather than on Github or Slack.

**Comment by jkotin** [comment]
Thank you all. Let's keep the image size the same as @gissoo suggests. Let's discuss the publication dates at our meeting. Even though I made the suggestion, I worry the extra information will be confusing. I'm eager to hear what everyone thinks.

**Comment by rlskoeser** [comment]
Updated to italicize and left-justify book titles and skip over cards without associated events unless the user is logged in.

Please test both logged in and not logged in - when navigating next/previous, for users who are not logged in it should skip over blank cards.

**Comment by thatbudakguy** [comment]
as per discussion at 12/4 mtg. "borrowed", "gifted", etc on mobile will become Borrow / Gift, and the "status" column will be changed to "activity".

**Comment by jkotin** [comment]
Looking at test site, logged-in on desktop.

Should I wait to test until the changes mentioned by @thatbudakguy happen? Somethings I notice now:

1. When the width of the screen is decreased on desktop, "borrow" becomes lowercase.  (I presume this happens because the phone format activates.)

2. The hyphen for a missing date in the "end date" is right justified. The hyphen should be centred. The dates should remain right justified.

For a later release:

1. Exploring the site now, I don't think we should include "type" in the member card view. We should stick with start date, title, and end date—i.e. only what's on the card.

2. We should have "BB [Brought Back]" in the end date column when the card indicates "BB" for the end date.

**Comment by rlskoeser** [comment]
@gissoo @thatbudakguy any opinion on centering hyphens for missing dates but keeping actual dates right justified? I think to implement we'll have to check if the date is missing and add a class to differentiate the styles.

**Comment by thatbudakguy** [comment]
@rlskoeser that class is already present, so this should be a snap. I'll add it.

**Comment by gissoo** [comment]
@rlskoeser on the card image viewer designs, the hyphens for missing dates follow the column alignment, which is left-aligned for start dates and right-aligned for end-dates. It would read better if they remain right-aligned. 

On the borrowing activities the only time the hyphens for missing dates are centered are in the start-date column where the dates are centered, it's better if it stays that way. It is only for the end-dates where they are right-aligned. 

On the membership activities they are always centered.

**Comment by gissoo** [comment]
@jkotin, regarding your second point about the hyphens being centered, we tested it now and it looks very messy and hard to read this way for columns that are **left or right-aligned**, because the centered hyphen will not match the alignment of any element in the column including the label, it will look like it's at the wrong place. I would suggest the hyphens will follow the alignment of the columns they are in for easier readability.

**Comment by jkotin** [comment]
@gissoo What about leaving the space blank? If that would also make the situation worse, then let's leave it right-aligned. Btw, I don't think this issue should be a blocker for this release.

**Comment by gissoo** [comment]
@jkotin The purpose of the hyphen was to make it explicit that data is missing. And also, I don't think leaving it blank makes enough of a difference to the eyes, but the effort of removing them from the design and implementation seems to be more than the effect in the current phase with the current priorities. I would suggest leaving it right-aligned for now, we can come back to this a bit later, I think there is something to be discussed regarding missing data and the why of missing data, and the goal of communicating it.

**Comment by jkotin** [comment]
@gissoo fair enough—I agree, thank you for exploring options with me.

**Comment by gissoo** [comment]
@jkotin Thank you! :)

**Comment by gissoo** [comment]
Designs now reflect what we agreed on regarding:
1. Left-aligning the title column and its label on the card image viewer on desktop and tablet  
2. Italicizing the title column on the card image viewer on desktop, tablet, and mobile. 
3. Changing the "Status" column to "Activity", and changing the content of the column from verbs to nouns and capitalizing it everywhere, e.g. "Borrowed" or "borrowed" to "Borrow" on card image viewer and borrowing activities pages on desktop, tablet and mobile.  

Here are the links:
1. Card image viewer
a. Desktop: "https://app.zeplin.io/project/5c48e7313d2b13bf7fc9fe14/screen/5db0b5030eb3b183b25b9200"
b. Tablet: "https://app.zeplin.io/project/5c48e7313d2b13bf7fc9fe14/screen/5db0b503f3d79e57b28a00e1"
c. Mobile: "https://app.zeplin.io/project/5c48e7313d2b13bf7fc9fe14/screen/5db0b50350731722a4809521"

2. Borrowing Activities
a. Desktop: "https://app.zeplin.io/project/5c48e7313d2b13bf7fc9fe14/screen/5cc1d1bb7cbe272d01d28b1e"
b. Tablet: "https://app.zeplin.io/project/5c48e7313d2b13bf7fc9fe14/screen/5cc216ebf11a741010f260d6"
c. Mobile: "https://app.zeplin.io/project/5c48e7313d2b13bf7fc9fe14/screen/5d0beb0e45c0e559e21c57ae"

**Comment by jkotin** [comment]
@gissoo The card image viewer windows look great. No changes to recommend. 

Are the designs for the borrowing activity pages out of scope for this particular issue? I think there are alignment issues there that should be revised. The author names should be left-aligned, the start dates left aligned, the end dates right aligned.

There is also an inconsistency between the card viewer page and borrowing activities page: one has "start-date" and "end-date," and one has "start date" and "end date." I don't have a strong preference as along as we are consistent. I think "start date" and "end date" are probably more technically correct -- i.e. no hyphens.

Should we change "publication date" to "pub. date" to save space on the desktop?

**Comment by jkotin** [comment]
The test site is not loading for me.  I get a "connection has timed out" message.  I'm off campus. I suspect we can close this issue, but I can't see the screens on the test site.

**Comment by jkotin** [comment]
@clmahoney you can close this if you can confirm that everything looks right. I can't access the test site at the moment and I'm travelling and then giving a talk tomorrow.

**Comment by clmahoney** [comment]
This all looks good to me!

**Comment by gissoo** [comment]
@jkotin Thanks for these points, I tested numbers 1 and 2 below and they work better. Thanks for catching number 3! I have updated the designs on Zeplin to reflect the following: 
1. Left-aligned authors on desktop and tablet – borrowing activities
2. left-aligned start dates on desktop and tablet – borrowing activities
3. Removing hyphens from "Start date" and "End date" on desktop and tablet – card image viewer

We had agreed to keep the "Publication Date" as is, because once we get to the "Books" section later there will need to be some consistencies. I would prefer to keep it as is right now. 

I talked to @rlskoeser about these changes. I will be making a chore on github, they will do it for this release if they can get to it, if not, I have made the changes in the design so it can be implemented in the future.
